File: Chapter 06/JS/section_6_3_2_graphql/fakeGraphQLHandler.js
/**
 * THE FAKE ENDPOINT.
 * Intercepts the HTTP call and returns our precise GraphQL JSON response.
 */
class FakeGraphQLHandler {
    /**
     * Simulates an asynchronous HTTP POST request.
     * @param {string} url - The endpoint being called.
     * @param {string} payload - The GraphQL query wrapped in JSON.
     * @returns {Promise<string>} A promise resolving to the precise JSON response.
     */
    async post(url, payload) {
        // The exact JSON response. Notice there is NO over-fetching here!
        const jsonResponse = `{
          "data": {
            "chipItem": { "name": "Salt & Vinegar Chips" },
            "sodaItem": { "price": 1.50 }
          }
        }`;
        return jsonResponse;
    }
}

module.exports = FakeGraphQLHandler;

File: Chapter 06/Python/section_6_3_2_graphql/fake_graphql_handler.py
import textwrap
class FakeGraphQLHandler:
    """THE FAKE ENDPOINT.
    
    Intercepts the HTTP call and returns our precise GraphQL JSON response.
    """
    
    def post(self, url: str, payload: str) -> str:
        """Simulates a synchronous HTTP POST request.
        
        Args:
            url (str): The mock GraphQL endpoint.
            payload (str): The JSON-encoded GraphQL query.
            
        Returns:
            str: A raw JSON string representing the exact requested data.
        """
        # The exact JSON response. Notice there is NO over-fetching here!
        return textwrap.dedent('''{
            "data": {
                "chipItem": { "name": "Salt & Vinegar Chips" },
                "sodaItem": { "price": 1.50 }
            }
        }''')